dev/lint-repository.sh

```diff
@@ -4,33 +4,25 @@
 
 # lint-commits.sh seeks to prevent the worsening of already present
 # problems, such as tab indentation in ml files. lint-repository.sh
-# seeks to prevent the (re-)introduction of solved problems, such as
-# newlines at the end of .v files.
+# also seeks to prevent the (re-)introduction of solved problems, such
+# as newlines at the end of .v files.
 
 CODE=0
 
-if [ -n "${TRAVIS_PULL_REQUEST}" ] && [ "${TRAVIS_PULL_REQUEST}" != false ];
-then
-    # skip PRs from before the linter existed
-    if [ -z "$(git ls-tree --name-only "${TRAVIS_PULL_REQUEST_SHA}" dev/lint-commits.sh)" ];
-    then
-        1>&2 echo "Linting skipped: pull request older than the linter."
-        exit 0
-    fi
-
-    # Some problems are too widespread to fix in one commit, but we
-    # can still check that they don't worsen.
-    CUR_HEAD=${TRAVIS_COMMIT_RANGE%%...*}
-    PR_HEAD=${TRAVIS_COMMIT_RANGE##*...}
-    MERGE_BASE=$(git merge-base "$CUR_HEAD" "$PR_HEAD")
-    dev/lint-commits.sh "$MERGE_BASE" "$PR_HEAD" || CODE=1
-fi
+# We assume that all merge commits are from the main branch
+# For Coq it is extremely rare for this assumption to be broken
```

Comment:
I considered that but then it seemed like I should add something for backport branches. It also seems kinda fragile.
Plus I'm not sure it would even give the right commit when someone merges master in their branch.
In the end I prefer keeping it simple and just accept that some per-commit checks will be missed if people put a merge in their PR. IMO this is fine since we essentially never do that, when we do it's a pathological case like #8778 and the per-commit checks are just not that important.

Reply:
It seems indeed fragile to me.